src/de/darmstadt/tu/crossing/CipherInputStream.cryptsl

```diff
@@ -0,0 +1,30 @@
+SPEC javax.crypto.CipherInputStream
+OBJECTS
+    byte[] b;
+    int off;
+    int len;
+    long n;
+    java.io.InputStream is;
+    javax.crypto.Cipher c;
+
+EVENTS
+    c1: CipherInputStream(is, c);
+
+    r1: read();
+    r2: read(b);
+    r3: read(b, off, len);
+    Reads := r1 | r2 | r3;
+
+    c: close();
+
+FORBIDDEN
+    CipherInputStream(is) => c1;
+
+ORDER
+    c1, Reads, c
```

Comment:
This should be Reads+.

Reply:
Ok, updated that, thanks. I guess, I should put Reads+ only in the CipherInputStream, because in the CipherOutputStream I don't need to change to Writes+, right?

Reply:
Hm, that depends on how these work. What happens if you call w1:write() on a `CipherOutputStream` twice? And what about first calling w2:write(byte[] b) and then w1:write()? Does the code throw an exception? If so, we should distinguish between the different write methods in the usage pattern.

Reply:
Yeah, I was thinking of that, if it is possible of calling write() two times or more. Still, has to be tested and seen. I will do that when writing the test cases for the rules